battle-service/src/main/java/com/company/battle/services/MoveService.java

```diff
@@ -1,19 +1,65 @@
 package com.company.battle.services;
 
-import com.company.common.dtos.MoveDto;
+import com.company.battle.repositories.GameRepository;
+import com.company.battle.repositories.MoveRepository;
+import com.company.battle.utils.Coordinate;
+import com.company.battle.utils.GameUtils;
+import com.company.common.models.GameEntity;
 import com.company.common.models.MoveEntity;
+import com.company.common.models.enums.GameStatus;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
 
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.UUID;
 
 @Service
+@RequiredArgsConstructor
 public class MoveService {
 
-    public MoveEntity add(MoveDto moveDto) {
-        return null;
+    @Autowired
+    private final MoveRepository moveRepository;
+
+    @Autowired
+    private final GameRepository gameRepository;
+
+    public MoveEntity add(MoveEntity moveEntity, UUID gameId) {
+
+        if (gameId != moveEntity.getGame().getId()) {
```

Comment:
you may ignore/override moveentity.gameId here . and remove this check.

Reply:
you should select game by gameId and check status .you should not work with nested entity (MoveEntity#game) here. 
due to it will be unclear for upstream classes that they need to construct whole entity before calling this method.